numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -652,6 +652,83 @@ static PyObject *
 }
 /**end repeat**/
 
+
+/*
+ * Convert string of bytes to a string representation
+ * much like bytes.__repr__, but convert all bytes
+ * (including ASCII) to the `\x00` notation.
+ *
+ * Largely copied from _Py_strhex_impl in CPython implementation
+ */
+static PyObject*
+_void_to_hex(const char* argbuf, const Py_ssize_t arglen)
+{
+    PyObject *retval;
+    Py_UCS1* retbuf;
+    Py_ssize_t i, j;
+
+#if defined(NPY_PY3K)
+#define extrachars 3
+#else
+#define extrachars 2
+#endif
+
+    assert(arglen >= 0);
+    if (arglen > PY_SSIZE_T_MAX / 4 - extrachars) {
+        return PyErr_NoMemory();
+    }
+
+    retbuf = (Py_UCS1*) PyMem_Malloc(extrachars + arglen*4);
+    if (!retbuf) {
+        return PyErr_NoMemory();
+    }
+
+    j = 0;
+#if defined(NPY_PY3K)
+    retbuf[j++] = 'b';
+#endif
+    retbuf[j++] = '\'';
+
+    for (i=0; i < arglen; i++) {
+        unsigned char c;
+        retbuf[j++] = '\\';
+        retbuf[j++] = 'x';
+        c = (argbuf[i] >> 4) & 0xf;
+        retbuf[j++] = Py_hexdigits[c];
+        c = argbuf[i] & 0xf;
+        retbuf[j++] = Py_hexdigits[c];
```

Comment:
This is lowercase - which makes separating the hex chars from the `x` a little harder on the eye